src/security/Handshake.h

```diff
@@ -75,6 +84,7 @@ class HandshakeParser
 
     bool resumingSession; ///< True if this is a resuming session
 
+    HandshakeType handshakeType; ///< hskClientHello or hskServerHello
```

Comment:
Probably I should not use the HandshakeType as type for HandshakeParser::handshakeType. It can be only "Client hello" or "ServerHello" and this is not the current handshake message type. A "Client Hello" (client request) or "ServerHello " (server response) may include many messages with different types.

I did not do this fix. But we may need to change the code to not use HandshakeType type here.

Reply:
I am rephrasing my previous comment here.
The HandshakeParser::handshakeType stores the info if we are parsing client hello or server hello. This is detected when we get a message of type HandshakeType::hskClientHello or a message of type HandshakeType::hskServerHello. These messages are the first messages in client request or in server response. So for the following:
> ... simplify the handshakeType assignment code, moving it away from hand-selected individual case statements to the top of parseHandshakeMessage()

I think it is ok to set  handshakeType when these messages detected inside case statement.

> We could also clear this field, after exiting from parseHandshakeMessage() but it is a bit awkward to do that in the current code

No we should not clear this field because it is true while we are parsing all included messages in server or client hello.

To avoid confusion with messages and messages types probably we should define a new type eg the HandshakeInitiator which has only the values  hClient or hServer. And also rename the handshakeType to handshakeInitiator.
Opinion on this?

Reply:
If we want to distinguish TLS clients from TLS servers, we should ask the Handshake parser creator to tell us which agent the handshake bytes are coming from instead of relying on raw input. Sending hskClientHello does not make a misbehaving TLS server a client!

How about adding a MessageSource { fromClient, fromServer } enum parameter to the Security::Handshake constructor, making that constructor explicit?

Reply:
commit 8b9544f6